src/components/Dashboard/UserBox.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+import classnames from 'classnames';
+import { useRouter } from 'next/router';
+
+import { useUserName } from '../AuthenticationProvider';
+import { useAuthentication } from '../AuthenticationProvider';
+import { Row } from '../Row/Row';
+import { Button } from '../Button/Button';
+import style from './UserBox.module.scss';
+
+export const UserBox = () => {
+  const [userBoxVisible, setUserBoxVisible] = useState(false);
+
+  const userName = useUserName();
+  const router = useRouter();
+  const { logout } = useAuthentication();
+
+  const handleRevealUserBox = () => {
+    setUserBoxVisible(!userBoxVisible);
+  };
+
+  const handleUpdate = () => {
+    router.push('/zmena');
+  };
+
+  return (
+    <Row
+      flexCol
+      justifyCenter
+      itemsCenter
+      className={classnames({
+        [style.userBoxOpen]: userBoxVisible,
+      })}
+    >
+      <button onClick={handleRevealUserBox} className={style.profileBox}>
+        <div className={style.profile} />
+        <div
+          className={classnames(style.arrow, {
+            [style.arrowOpen]: userBoxVisible,
+          })}
+        />
+      </button>
+      {userBoxVisible && (
+        <>
+          <h2>{userName}</h2>
+          <Button
+            type="button"
+            className={style.buttonUpdate}
+            label="upravit profil"
+            onClick={handleUpdate}
+          />
+          <Button
+            type="button"
+            className={style.buttonLogOut}
+            label="odhlásit"
+            onClick={logout}
+          />
```

Comment:
It is a more common practice to pass the label as a child. Like so

![CleanShot 2023-12-06 at 22 15 13](https://github.com/KamilaSemotanova/tracker/assets/12488648/0563149a-2b62-4c50-a596-94115b62ca31)

Since it is more common, you create a button API that obeys the principle of "least surprises." Also, it makes the button more flexible; you can easily add icons to it and do some nice styling. Doing that in the prop is possible but awkward